test/java/org/opendatakit/briefcase/export/ExportConfigurationTest.java

```diff
@@ -171,6 +171,19 @@ public void resolves_if_we_need_to_pull_depending_on_a_pair_or_fields() {
     assertThat(config.setPullBefore(false).setPullBeforeOverride(DONT_PULL).resolvePullBefore(), is(false));
   }
 
+  @Test
+  public void resolves_if_we_need_to_export_media_files_depending_on_a_pair_or_fields() {
+    assertThat(empty().resolveExportMedia(), is(true));
+    assertThat(empty().setExportMedia(true).resolveExportMedia(), is(true));
+    assertThat(empty().setExportMedia(false).resolveExportMedia(), is(false));
+    assertThat(empty().setExportMedia(true).setExportMediaOverride(ExportMediaOverrideOption.INHERIT).resolveExportMedia(), is(true));
```

Comment:
Forgive my repetition, and I accept that you aren’t going to change it, but
`assertTrue` would tell me right away what kind of assert this is without having to read all the way to the end for the `is(true)`. This is verbose for no good reason.

Reply:
It doesn't hurt to remind why I want to follow the `assertThat` style of assertions, especially in a case like this one, which is clearly a tradeoff, as you point out:

- I find that the classic Junit4 assertions have some issues:
  - It's common to make mistakes in the order of arguments, by swapping the expected value with the actual value
  - You need to know about a range of different `assert*` methods in advance
- With the `assertThat` style assertions:
  - Assertions are like an English phrase (subject and predicate), which is easier to understand
  - All assertions follow the same structure, have the same arguments, in the same position, and they always mean the same thing.
  - The error messages are more expressive and useful
  - The matchers are composable and you can easily make your own!
    - This is, by far, the best feature. 
    - It means that you can adapt your matchers to your problem's domain
    - It also means that you can break your assertion into smaller & simpler assertions and combine them without making the resulting assertion harder to understand.
    - The error messages get composed as well, producing useful errors that give the much-needed context while fixing something.

The most common tradeoff is that sometimes you could replace your assertion with an `assertTrue` or an `assertNull`, which only takes an argument but this is outweighed by the benefits of being able to express complex, domain related assertions in the long run. 

This is more of a strategic design decision than a practical, down-to-the-code decision. This assertion style also plays well with test scenarios like the ones we're planning to add to JavaRosa, which help to make the tests about asserting the program's behavior.